### ISSUE TITLE
Wrong pixel to scene mapping in the cocoa-based viewer

### ISSUE THREAD

**Issue by agheata** [root-project/root#12035]
- [x] Checked for duplicates

### Describe the bug

The GL geometry viewer allows "picking" the center of view for a drawn geometry. This is not working correctly on Mac.

### Expected behavior

In the GL geometry viewer, the tab "Guides/Camera center" allows to "Pick center" with the mouse, which will change the default center of view to the 2D mouse coordinates of the clicked point on the 3D coordinates of the drawn model. The "depth" is chosen such that the point on the drawn object surface below the mouse is picked.

### To Reproduce

Run the macro `$ROOTSYS/tutorials/geom/rootgeom.C
root[0] gGeoManager->GetVolume("R")->Draw("ogl");
Rotate the drawn "R", then go to tab "Guides" and select [Camera center]/Show
A marker should appear in the default center of view.
Now click on "Pick center", then click somewhere on the drawn letter "R"
On linux, the center of view moves on the surface of "R" just below the mouse, and you can freely rotate about this
On mac, the center moves elsewhere than below the mouse position, showing that the device coordinates are wrongly mapped onto the scene coordinates. 

### Setup

Any of the recent releases, tested on master
OS: linux Ok, mac buggy

### Additional context

See: https://root-forum.cern.ch/t/mouse-position-shifts-in-geometry-display/52937/4

**Comment by couet** [comment]
I am not sure it is a Cocoa issue as it is the GL window.  Maybe @TimurP or @osschar  have an idea about it ?

**Comment by osschar** [comment]
Is the mac position scaled by 1/2 or 2, ie, the retina scale factor?

**Comment by couet** [comment]
I am not sure what good behavior should be. @agheata might know if that's the factor

**Comment by agheata** [comment]
> Is the mac position scaled by 1/2 or 2, ie, the retina scale factor

No idea, but shouldn't this get detected automatically? I'm reading that "You need to use [glfwGetFramebufferSize](http://www.glfw.org/docs/latest/group__window.html#ga0e2637a4161afb283f5300c7f94785c9) to get the actual size of the window in pixels on displays that use window coordinates that don't match pixels. Then pass those values to glViewport

**Comment by agheata** [comment]
#12055 fixes the problem of drawing pixels as dot markers in the pad

I tracked the mouse-picking problem down to the method `TGLEventHandler::HandleButton` which is called after clicking `Pick center` and then clicking to select the center in the image. The scaling problem happens while calling:
```
            TGLVector3 v(event->fX, event->fY, 0.5*fGLViewer->fSelRec.GetMinZ());
            fGLViewer->CurrentCamera().WindowToViewport(v);
            v = fGLViewer->CurrentCamera().ViewportToWorld(v);
```
 This is support to convert the mouse position window coordinates to GL world coordinates. However, I do not understand where the screen scaling has to be applied.

**Comment by osschar** [comment]
It should probably be applied to scale event->fX, event->fY ... or in XXCamera::WindowToViewport(v). I'll take a look, i think I can get my hands on a mac here.

**Comment by osschar** [comment]
Hmmh, on the other hand ... this used to work for 10+ years ... so something must have changed in how event coordinates are reported. @agheata were you able to reproduce it? On what version of osx? Does it go away if one chooses a non-retina display mode?

**Comment by osschar** [comment]
Was this a Cocoa build? Do we even still support X11 on macs?

**Comment by agheata** [comment]
> Hmmh, on the other hand ... this used to work for 10+ years ... so something must have changed in how event coordinates are reported. @agheata were you able to reproduce it? On what version of osx? Does it go away if one chooses a non-retina display mode?

Yes, I can reproduce on my MacBook Air (M1, 2020) with Monterey and a Cocoa build, no matter the scaled display mode. I'm not sure the problem is that recent. I observed in addition that the picked coordinates are exactly half the correct ones. The same factor of two affected the raytracing part of the problem that I managed to fix in #12055

**Comment by agheata** [comment]
This will be fixed by #12088. Thanks to @TimurP for the hint on how to retrieve the scaling factor in TGLUtil
